ase/scripts/generate_ase_environment.py

```diff
@@ -387,10 +387,11 @@ def config_qsys_sources(filelist, vlog_srcs):
             print("Preparing {0}:".format(q))
             print("  Copying {0} to {1}...".format(src_dir, tgt_dir))
             try:
-                shutil.copytree(src_dir, tgt_dir)
-            except Exception:
-                errorExit("Failed to copy tree {0} to {1}".format(src_dir,
-                                                                  tgt_dir))
+                shutil.copytree(src_dir, tgt_dir, symlinks=False, ignore=None)
+            except Exception as e:
+                errorExit("""Failed to copy tree {0} to \
```

Comment:
I don't think the format string here works the way you think it does. I think it will have whitespace before {1}. More likely, what you want is:

```
ErrorExit("Failed to copy tree {0} to"
          "{1}: Exception = {2}".format(src_dir, tgt_dir, e))
```

The Python limit of 80 columns would have been good style in 1980.